shub_image/deploy.py

```diff
@@ -125,9 +127,12 @@ def _prepare_deploy_params(project, version, image_name, endpoint, apikey,
     if not username:
         params['pull_insecure_registry'] = True
     else:
+        # Missing password leads to auth request to registry authentication
+        # service without 'account' query parameter which breaks login
+        # procedure.
         params['pull_auth_config'] = json.dumps(
             {'username': username,
-             'password': password,
+             'password': password or ' ',
```

Comment:
Works as well (you can try by yourself by marking one of your repo as private in settings).  For private Docker hub images rules are the same as for any other private registry: you must provide username & password or there will be `image not found` error.